Automattic/newspack-blocks
# chore: update dependencies; slider JS code changes for major version update

### All Submissions:






### Changes proposed in this Pull Request:



* Updates as many NPM packages as possible to match https://github.com/Automattic/newspack-plugin/pull/1135.
* Implements Dependabot for automated incremental package updates.
* Adds a missing `postcss.config.js` file which was causing PostCSS to not process any CSS files.
* Implements [required code changes](https://swiperjs.com/migration-guide) for the major SwiperJS update to version 7.

~This PR depends on https://github.com/Automattic/newspack-plugin/pull/1135 as well as a new released version of Newspack Components to update its own dependencies. Until this happens, the build CI jobs in this PR will fail. In the meantime, this PR can be tested locally if your test site also has that branch of Newspack Plugin checked out.~ No longer needed, as the PR has been merged and Newspack Components has been updated on NPM.

### How to test the changes in this Pull Request:

Can be tested locally:

1. ~(Until Newspack Components is updated in NPM) Check out https://github.com/Automattic/newspack-plugin/pull/1135 in Newspack Plugin. In that plugin, go to `assets/components` and run `npm run prepublishOnly` to build the components package.~ No longer needed, as Newspack Components has been updated on NPM.
2. Using [NVM](https://github.com/nvm-sh/nvm), set your current Node version to v16.11.1 (matching Calypso's supported version)
3. Ensure `npm ci` runs
4. Ensure all NPM build, lint and test scripts still run (`build`, `dev`, `lint`)
5. Test all blocks in editor and on front-end, especially Posts Carousel in AMP and non-AMP mode (which uses SwiperJS). There should be no functional changes compared to `master`.

### Other information:
